File: 1502/go/1502.go
func canMakeArithmeticProgression(arr []int) bool {
    sort.Ints(arr)
    commDiff := arr[0] - arr[1]
    n := len(arr)

    for i := 1; i < n-1; i++ {
        if (arr[i] - arr[i+1]) != commDiff {
            return false
        }
    }

    return true
}

File: 1502/js/1502.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var canMakeArithmeticProgression = function(arr) {
    arr.sort((a, b) => a - b);
    let n = arr.length
    let commDiff = arr[0] - arr[1]

    for (let i = 1; i < n-1; i++) {
        if (commDiff != (arr[i] - arr[i+1])) {
            return false;
        }
    }

    return true;
};